pkg/agent/memberlist/cluster_test.go

```diff
@@ -0,0 +1,631 @@
+			count += i
+		}
+		return count
+	}(egressNum)
+
+	genNodes := func(n int) []string {
+		nodes := make([]string, n)
+		for i := 0; i < n; i++ {
+			nodes[i] = fmt.Sprintf("node-%d", i)
+		}
+		return nodes
+	}
+
+	nodes := genNodes(12)
+	testCases := []struct {
+		name           string
+		nodes          []string
+		consistentHash *consistenthash.Map
+	}{
+		{
+			name:           fmt.Sprintf("assign owner Node for %d Egress", egressNum),
+			nodes:          nodes[:10],
+			consistentHash: consistenthash.New(defaultVirtualNodeReplicas, nil),
+		},
+		{
+			// Failover, when Node failed, Egress should move to available Node.
+			name:           "a Node fail then Egress should move",
+			nodes:          nodes[1:10],
+			consistentHash: consistenthash.New(defaultVirtualNodeReplicas, nil),
+		},
+		{
+			// Egress should move when Node added in cluster.
+			name:           "add new Node then Egress should move",
+			nodes:          nodes[:11],
+			consistentHash: consistenthash.New(defaultVirtualNodeReplicas, nil),
+		},
+		{
+			name:           fmt.Sprintf("recover to %d nodes", 10),
+			nodes:          nodes[:10],
+			consistentHash: consistenthash.New(defaultVirtualNodeReplicas, nil),
+		},
+	}
+
+	nodeSelectedForEgress := func(consistentHash *consistenthash.Map, egressName, nodeName string) bool {
+		return consistentHash.Get(egressName) == nodeName
+	}
+	for _, testC := range testCases {
+		t.Run(testC.name, func(t *testing.T) {
+			nodes := testC.nodes
+			testC.consistentHash.Add(nodes...)
+			nodeEgress := make(map[string][]string, len(nodes))
+			hitCount := 0
+			seqSum := 0
+			for i := 0; i < egressNum; i++ {
+				egressName := fmt.Sprintf("%d", i)
+				for j := range nodes {
+					localNode := nodes[j]
+					if nodeSelectedForEgress(testC.consistentHash, egressName, localNode) {
+						nodeEgress[localNode] = append(nodeEgress[localNode], egressName)
+						hitCount++
+						seqSum += i
+					}
+				}
+			}
+			assert.Equal(t, expectEgressSeqSum, seqSum, "Egress seq")
+			for _, node := range nodes {
+				t.Logf("Node (%s) Egress: %#v", node, nodeEgress[node])
+			}
+			assert.Equal(t, egressNum, hitCount, "Egress total num should be 30")
+		})
+	}
+}
+
+func TestCluster_ShouldSelectEgress(t *testing.T) {
+	testCases := []struct {
+		name         string
+		nodeNum      int
+		egressIP     string
+		expectedNode string
+	}{
+		{
+			name:         "select Node from alive nodes",
+			nodeNum:      0,
+			egressIP:     "1.1.1.1",
+			expectedNode: "",
+		},
+		{
+			name:         "select Node from alive nodes",
+			nodeNum:      1,
+			egressIP:     "1.1.1.1",
+			expectedNode: "node-0",
+		},
+		{
+			name:         "select Node from alive nodes",
+			nodeNum:      3,
+			egressIP:     "1.1.1.1",
+			expectedNode: "node-1",
+		},
+		{
+			name:         "select Node from alive nodes",
+			nodeNum:      10,
+			egressIP:     "1.1.1.1",
+			expectedNode: "node-1",
+		},
+		{
+			name:         "select Node from alive nodes",
```

Comment:
The name is used to figure out which subtest fails quickly and give a brief description of the tests. It's pointless if they are same.
You can use "select Node from 0 node", "select Node from 1 node", "select Node from 3 nodes" etc.

Reply:
done, 
I have labeled them in t.Run before. It should have the same effect I think.
t.Run(fmt.Sprintf("%s-nodeNum-%d", tCase.name, tCase.nodeNum), func(t *testing.T)